File: Typescript/Objects/Estudiante.ts
export class Estudiante{
    readonly nombre: string;
    readonly edad: number;
    private promedio: number;

    constructor(nombre: string, edad: number, promedio: number){
        this.nombre = nombre;
        this.edad = edad;
        this.promedio = promedio;
    }

    public mostrarInfo(): void{
        console.log(`Nombre: ${this.nombre}\nEdad: ${this.edad},\nPromedio: ${this.promedio}\n`);
    }

    public setPromedio(promedio: number){
        this.promedio = promedio;
    }

    public getPromedio():number{
        return this.promedio;
    }
}

File: Kotlin/src/Objects/Estudiante.kt
package Objects;

class Estudiante(val nombre: String, val edad: Int, private var promedio: Double){
    constructor(nombre: String, edad: Int): this(nombre, edad,0.0){}
    constructor(nombre: String):this(nombre,0,0.0){}

    fun mostrarInfo(){
        println("Nombre: ${this.nombre}")
        println("Edad: ${this.edad}")
        println("Promedio: ${this.promedio}\n")
    }

    fun getPromedio(): Double{
        return this.promedio
    }

    fun setPromedio(value: Double){
        this.promedio = value
    }
}